test/unit/duration-type-tests.js

```diff
@@ -124,6 +128,37 @@ describe("Duration", function () {
             });
         });
     });
+    describe("constructor type validation", function () {
+        it("should create duration type correctly", function () {
+            let keys = [0, Long.ZERO, 123, BigInt(0)];
+            keys.forEach(function (item) {
+                new Duration(0, 0, item);
+            });
+            new Duration(0, 0, 0);
+        });
+    });
+    describe("constructor type validation (invalid types)", function () {
+        it("should throw error when creating from invalid types", function () {
+            let ns_keys = [
+                "0",
+                0xfffffffffffffffff,
+                { _: 1 },
+                [0],
+                BigInt("213769213769213769213769"),
+            ];
+            ns_keys.forEach(function (item) {
+                assert.throws(() => {
+                    new Duration(0, 0, item);
+                }, Error);
```

Comment:
Maybe also check if the correct error is thrown?

Reply:
no.